optimism/src/keccak/environment.rs

```diff
@@ -47,18 +53,16 @@ pub struct KeccakEnv<F> {
     /// Byte-length of the 10*1 pad (<=136)
     pub(crate) pad_len: u64,
 
-    /// Precomputed 2^pad_len (including 2^0 = 1)
-    two_to_pad: [F; RATE_IN_BYTES + 1],
-    /// Precomputed inverses of the padding lengths (including dummy 0)
-    inv_pad_len: [F; RATE_IN_BYTES + 1],
+    /// Precomputed 2^pad_len
+    two_to_pad: [F; RATE_IN_BYTES],
     /// Precomputed suffixes for the padding blocks
-    pad_suffixes: [[F; PAD_SUFFIX_LEN]; RATE_IN_BYTES + 1],
```

Comment:
Why do `pad_suffixes` and `two_to_pad` get one less element than they had before? `RATE_IN_BYTES` instead of `RATE_IN_BYTES + 1`?

Reply:
Because before, the constraints for the dummy witness set to all zeros should also hold. But now that you can select to run them or not, you can be fine with simply the "real" values. The one with dummy entries will never be used now.